destination/destination_integration_test.go

```diff
@@ -0,0 +1,170 @@
+	is := is.New(t)
+	messageBody := "Test message body"
+
+	client, url, cfg, err := prepareIntegrationTest(t)
+	ctx := context.Background()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	destination := &Destination{}
+	err = destination.Configure(ctx, cfg)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = destination.Open(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	metadata := sdk.Metadata{}
+	record := sdk.Util.Source.NewRecordCreate(
+		sdk.Position("111111"),
+		metadata,
+		sdk.RawData("1111111"),
+		sdk.RawData(messageBody),
+	)
+
+	ret, err := destination.Write(ctx, []sdk.Record{record})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	is.Equal(ret, 1)
+	// wait a little bit after we send message, it wont show up right away
+	time.Sleep(30 * time.Second)
+
+	message, err := client.ReceiveMessage(
+		context.Background(),
+		&sqs.ReceiveMessageInput{
+			QueueUrl: url.QueueUrl,
+		},
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(message.Messages) == 0 {
+		t.Fatalf("expected 1 message. Got %d", len(message.Messages))
+	}
+
+	is.Equal(*message.Messages[0].Body, messageBody)
+
+	_ = destination.Teardown(ctx)
+}
+
+func prepareIntegrationTest(t *testing.T) (*sqs.Client, *sqs.GetQueueUrlOutput, map[string]string, error) {
+	cfg, err := parseIntegrationConfig()
+	if err != nil {
+		t.Skip(err)
+	}
+
+	client, err := newAWSClient(cfg)
+	if err != nil {
+		t.Fatalf("could not create S3 client: %v", err)
+	}
+
+	sourceQueue := "test-queue-destination-" + uuid.NewString()
+
+	_, err = client.CreateQueue(context.Background(), &sqs.CreateQueueInput{
+		QueueName: &sourceQueue,
+	})
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	queueInput := &sqs.GetQueueUrlInput{
+		QueueName: &sourceQueue,
+	}
+	// Get URL of queue
+	urlResult, err := client.GetQueueUrl(context.Background(), queueInput)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	t.Cleanup(func() {
+		err := deleteSQSQueue(t, client, urlResult.QueueUrl)
+		if err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	cfg[ConfigKeyAWSQueue] = sourceQueue
+
+	return client, urlResult, cfg, nil
+}
+
+func deleteSQSQueue(t *testing.T, svc *sqs.Client, url *string) error {
+	_, err := svc.DeleteQueue(context.Background(), &sqs.DeleteQueueInput{
+		QueueUrl: url,
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func newAWSClient(cfg map[string]string) (*sqs.Client, error) {
+	awsConfig, err := config.LoadDefaultConfig(context.Background(),
+		config.WithCredentialsProvider(
+			credentials.NewStaticCredentialsProvider(
+				cfg[ConfigKeyAWSAccessKeyID],
+				cfg[ConfigKeyAWSSecretAccessKey],
+				cfg[ConfigKeyAWSToken])),
+	)
+	if err != nil {
+		return nil, err
+	}
+	// Create a SQS client from just a session.
+	sqsClient := sqs.NewFromConfig(awsConfig)
+
+	return sqsClient, nil
+}
+
+func parseIntegrationConfig() (map[string]string, error) {
+	awsAccessKeyID := os.Getenv("AWS_ACCESS_KEY_ID")
```

Comment:
Nice, now we just need to set up some keys in the GitHub repo so that the CI actually executes these tests. You can ask @maha-hajja for help if you need it, she did it in the S3 connector.

Reply:
Sure! What AWS account can I use for github tests? Would just staging one be ok?

Reply:
@anna-cross yes we usually use Staging

Reply:
@hariso I don't think it was to be running all the time, the SQS Queue is created when the test starts and is cleaned up afterwards 🤔 (unless im missing something)

Reply:
OK, so we're **not** paying only if there are queues (never used SQS before)? i.e. there's no SQS service running all the time.

Reply:
Yup, and the queue are cleaned up, so there shouldn't be an issue of paying for this